src/v2/providers/schedule.ts

```diff
@@ -0,0 +1,198 @@
+    return {
+      schedule: args,
+      opts: {},
+    };
+  }
+  const scheduleArgs: Partial<ScheduleArgs> = {
+    schedule: args.schedule,
+    timeZone: args.timeZone,
+    retryCount: args.retryCount,
+    maxRetryDuration: args.maxRetryDuration,
+    minBackoffDuration: args.minBackoffDuration,
+    maxBackoffDuration: args.maxBackoffDuration,
+    maxDoublings: args.maxDoublings,
+  };
+  const opts = { ...args };
+  delete (opts as any).schedule;
+  delete (opts as any).timeZone;
+  delete (opts as any).retryCount;
+  delete (opts as any).maxRetryDuration;
+  delete (opts as any).minBackoffDuration;
+  delete (opts as any).maxBackoffDuration;
+  delete (opts as any).maxDoublings;
+
+  scheduleArgs.opts = opts;
+  return scheduleArgs as ScheduleArgs;
+}
+
+/** The Cloud Function type for Schedule triggers. */
+export interface ScheduleFunction extends HttpsFunction {
+  __requiredAPIs?: ManifestRequiredAPI[];
+  run(data: express.Request): any | Promise<any>;
+}
+
+/** Options that can be set on a Schedule trigger. */
+export interface ScheduleOptions extends options.GlobalOptions {
+  /** The schedule, in Unix Crontab or AppEngine syntax. */
+  schedule: string;
+
+  /** The timezone that the schedule executes in. */
+  timeZone?: timezone;
+
+  /** The number of retry attempts for a failed run. */
+  retryCount?: number;
+
+  /** The time limit for retrying. */
+  maxRetryDuration?: string;
+
+  /** The minimum time to wait before retying. */
+  minBackoffDuration?: string;
+
+  /** The maximum time to wait before retrying. */
+  maxBackoffDuration?: string;
+
+  /** The time between will double max doublings times. */
+  maxDoublings?: number;
+}
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param schedule - The schedule, in Unix Crontab or AppEngine syntax.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  schedule: string,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction;
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param options - Options to set on scheduled functions.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  options: ScheduleOptions,
+  handler: (req: express.Request) => any | Promise<any>
+): ScheduleFunction;
+
+/**
+ * Handler for scheduled functions. Triggered whenever the associated
+ * scheduler job sends a http request.
+ * @param args - Either a schedule or an object containing function options.
+ * @param handler - A function to execute when triggered.
+ * @returns A function that you can export and deploy.
+ */
+export function onSchedule(
+  args: string | ScheduleOptions,
+  handler: (req: express.Request) => any | Promise<any>
```

Comment:
Should `handler` signature be a little more specific? e.g. `(req ...) => void | Promise<void>`

Reply:
Yeah, void makes sense in this context